Src/Plugins/Security/SEC101_054.AlibabaAccessKeyValidator.cs

```diff
@@ -0,0 +1,124 @@
+    {
+        internal static AlibabaAccessKeyValidator Instance;
+
+        private const string KeyNotFound = "InvalidAccessKeyId.NotFound";
+        private const string InvalidSecret = "SDK.InvalidAccessKeySecret";
+        private const string ProductKeyInvalidFormat = "The productKey format is incorrect.";
+
+        static AlibabaAccessKeyValidator()
+        {
+            Instance = new AlibabaAccessKeyValidator();
+        }
+
+        public static string IsValidStatic(ref string matchedPattern,
+                                           ref Dictionary<string, string> groups,
+                                           ref string failureLevel,
+                                           ref string fingerprint,
+                                           ref string message)
+        {
+            return IsValidStatic(Instance,
+                                 ref matchedPattern,
+                                 ref groups,
+                                 ref failureLevel,
+                                 ref fingerprint,
+                                 ref message);
+        }
+
+        public static string IsValidDynamic(ref string fingerprint, ref string message)
+        {
+            return IsValidDynamic(Instance,
+                                   ref fingerprint,
+                                   ref message);
+        }
+
+        protected override string IsValidStaticHelper(ref string matchedPattern,
+                                                      ref Dictionary<string, string> groups,
+                                                      ref string failureLevel,
+                                                      ref string fingerprintText,
+                                                      ref string message)
+        {
+            if (!groups.TryGetNonEmptyValue("account", out string account) ||
+                !groups.TryGetNonEmptyValue("password", out string password))
+            {
+                return nameof(ValidationState.NoMatch);
+            }
+
+            fingerprintText = new Fingerprint()
+            {
+                Platform = nameof(AssetPlatform.AlibabaCloud),
+                Account = account,
+                Password = password,
+            }.ToString();
+
+            return nameof(ValidationState.Unknown);
+        }
+
+        protected override string IsValidDynamicHelper(ref string fingerprintText,
+                                                       ref string message)
+        {
+            var fingerprint = new Fingerprint(fingerprintText);
+            string account = fingerprint.Account;
+            string password = fingerprint.Password;
+
+            try
+            {
+                // Taken from https://www.alibabacloud.com/help/doc-detail/63638.htm
+                IClientProfile clientProfile = DefaultProfile.GetProfile("cn-shanghai", account, password);
+                var client = new DefaultAcsClient(clientProfile);
```

Comment:
>DefaultAcsClient [](start = 33, length = 16)

Hey, is this type disposable? If so put a 'using' in front of it.

Reply:
Eddy made the same observation.  None are disposable.

---
In reply to: [592687256](https://github.com/microsoft/sarif-pattern-matcher/pull/285#discussion_r592687256) [](ancestors = 592687256)